extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/VariantTemplateProducer.java

```diff
@@ -166,15 +172,15 @@ public TemplateVariants(Map<Variant, String> variants, String defaultTemplate) {
     }
 
     static String parseMediaType(String suffix) {
-        // TODO support more media types...
-        if (suffix.equalsIgnoreCase(".html")) {
-            return "text/html";
+        // TODO we need a proper way to parse the media type
+        if (suffix.equalsIgnoreCase(".html") || suffix.equalsIgnoreCase(".htm")) {
+            return Variant.TEXT_HTML;
         } else if (suffix.equalsIgnoreCase(".xml")) {
-            return "text/xml";
+            return Variant.TEXT_XML;
         } else if (suffix.equalsIgnoreCase(".txt")) {
-            return "text/plain";
+            return Variant.TEXT_PLAIN;
         } else if (suffix.equalsIgnoreCase(".json")) {
-            return "application/json";
+            return Variant.APPLICATION_JSON;
```

Comment:
We support JSON but I didn't see any escaper for JSON.

Reply:
Well, actually we don't support JSON ATM ;-). TBD